ui-performance.md

```diff
@@ -0,0 +1,470 @@
+
+## The performance overlay
+
+<div class="whats-the-point" markdown="1">
+
+<b> <a id="whats-the-point" class="anchor" href="#whats-the-point" aria-hidden="true"><span class="octicon octicon-link"></span></a>What's the point?</b>
+
+* Turn on the overlay with the Flutter inspector, at the command
+  line, or programmatically.
+* Observe the app's performance graphically.
+* One graph displays performance of Dart code. The other displays
+  performance of the graphics engine.
+* The overlay displays the last 300 frames for each engine.
+
+</div>
+
+The
+[PerformanceOverlay](https://docs.flutter.io/flutter/widgets/PerformanceOverlay-class.html)
+widget, from the
+[widget library](https://docs.flutter.io/flutter/widgets/widgets-library.html),
+displays performance statistics using two graphs.
+The following screenshot shows the performance overlay running on the Flutter
+Gallery example:
+
+<center><img src="images/performance-overlay-green.png" alt="screenshot of performance overlay showing zero jank"></center>
+<center>The graphs produced by the performance overlay.<br>THe vertical green bar represents the current frame.</center>
+
+Each graph represents the last 300 frames for the following two threads used in every Flutter
+app:
+
+* Engine<br>
+  The engine graph (on the bottom) represents time spent executing Dart code in the Dart VM.
+  This thread is sometimes called the UI thread and includes code that you wrote,
+  and code performed by Flutter’s framework on your app’s behalf.
+
+* Rasterizer<br>
+  The rasterizer graph (on the top) represents time spent painting your application,
+  or interfacing with the phone’s GPU (graphics processing unit) thread. When your app creates
+  and displays a scene, the framework creates a “layer tree”, a lightweight object containing
+  device-agnostic painting commands, and sends the layer tree to the GPU thread to be rendered
+  on the device. You cannot directly access the GPU thread or its data but, if this thread is
+  slow, it’s a result of something you’ve done in the Dart code.
+
+Generally, each frame should be created and displayed within 16ms (or 1/60th of a second).
+A frame exceeding this limit fails to display, resulting in jank, and a vertical red bar
+appears in one or both of the graphs. If a red bar appears in the engine graph,
+the Dart code is too expensive. If a red vertical bar appears in the rasterizer graph,
+the scene is too complicated to render quickly.
+
+<center><img src="images/performance-overlay-jank.png" al="Screenshot of performance overlay showing jank."></center>
+<center>The vertical red bar in the rasterier graph indicates that the current frame is expensive to paint.</center>
+
+### Displaying the performance overlay
+
+You can toggle the performance overlay in several ways:
+
+* Using the Flutter Inspector
+* From the command line
+* Programmatically
+
+Flutter Inspector is available in the VSCode, Android Studio, and IntelliJ plugins,
```

Comment:
The Flutter Inspector...

Reply:
You can strike VSCode here - the long term plan is to have it available, but not near term.